django/contrib/staticfiles/handlers.py

```diff
@@ -52,6 +54,12 @@ def get_response(self, request):
         except Http404 as e:
             return response_for_exception(request, e)
 
+    async def get_response_async(self, request):
+        try:
+            return await sync_to_async(self.serve)(request)
+        except Http404 as e:
+            return response_for_exception(request, e)
```

Comment:
`return await sync_to_async(response_for_exception)(request, e)`?

Reply:
Can we add tests for catching 404s?

Reply:
We wrap `response_for_exception` in an async context elsewhere:

https://github.com/django/django/blob/f47d5aac622c334ebeba06b7460204aeb98661e2/django/core/handlers/exception.py#L34-L51